### ISSUE TITLE
fetcher errors (TypeError, invalid-json)

### ISSUE THREAD

**Issue by vwegert** [deg0nz/MMM-PublicTransportBerlin#66]
After pulling the December changes, I'm seeing the following error multiple times right after MM2 startup:

```
[ERROR] Error initializing fetcher: TypeError: Cannot read property 'getStationName' of undefined
```

When the UI is loaded, the following errors appear:

```
[ERROR] FetchError: invalid json response body at https://bvg-apps-ext.hafas.de/bin/mgate.exe? reason: Unexpected end of JSON input
    at /home/magicmirror/MagicMirror/modules/MMM-PublicTransportBerlin/node_modules/node-fetch/lib/index.js:273:32
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async BvgFetcher.getStationName (/home/magicmirror/MagicMirror/modules/MMM-PublicTransportBerlin/BvgFetcher.js:27:19)
    at async Class.createFetcher (/home/magicmirror/MagicMirror/modules/MMM-PublicTransportBerlin/node_helper.js:22:27) {
  type: 'invalid-json'
}
```

**Comment by ahaupt** [comment]
Also seeing this, but I'd rather think it's a problem with the BVG api service. See example from https://v5.bvg.transport.rest/api.html

```
[notebook] ~ % curl 'https://v5.bvg.transport.rest/locations?query=alexanderplatz&results=1'
{
	"message": "invalid json response body at https://bvg-apps-ext.hafas.de/bin/mgate.exe? reason: Unexpected end of JSON input",
	"type": "invalid-json",
	"error": true,
	"msg": "invalid json response body at https://bvg-apps-ext.hafas.de/bin/mgate.exe? reason: Unexpected end of JSON input"
}
```

**Comment by mazim-co** [comment]
Same here. Getting the same error above. Any ideas how to fix this?

**Comment by deg0nz** [comment]
Thanks for reporting. This looks like an error that is out of our reach. I wasn't able to find an issue ticket in `hafas-client`.

I will jut ping @derhuerst here, maybe he can help us to narrow this error down.

**Comment by derhuerst** [comment]
AFAIK this is due to the user agent that is being passed into `hafas-client` during initialisation. It seems like BVG or HaCon blocks certain User-Agents, because https://v5.bvg.transport.rest was down and I fixed it by changing the user agent.

**Comment by KristjanESPERANTO** [comment]
My PR I improves the situation. But I still get in rare cases the error message `FetchError: invalid json response body at https://bvg-apps-ext.hafas.de/bin/mgate.exe? reason: Unexpected end of JSON input`. Nevertheless, departures are displayed with this fix.

I suggest not closing this issue yet, even if my PR has been merged. It would be nice if we could find out the cause of the error message.

**Comment by deg0nz** [comment]
This issue was closed because of merging #67. Please feel free to open again of you encounter any errors
